### ISSUE TITLE
easyocr is useless in demo, why use it?

### ISSUE THREAD

**Issue by MOSSV2** [microsoft/OmniParser#5]

**Comment by nmstoker** [comment]
Do you have an alternative recommendation?

**Comment by aliencaocao** [comment]
I agree that it is not the best OCR solution, and also often doesn't get the text right for me. I recommend PaddleOCR https://github.com/PaddlePaddle/PaddleOCR/blob/main/README_en.md

**Comment by nmstoker** [comment]
Thanks @aliencaocao . Looks like it only needs modest changes to integrate a different OCR into utils.py, eg here:

https://github.com/microsoft/OmniParser/blob/d7708e830a2f16a40a8970211a9911fa776d4aa2/utils.py#L375

Is that correct and was it straightforward when you switched out EasyOCR for the demo above?

**Comment by aliencaocao** [comment]
Yes, I can make a PR if you wish, I already have it impl in my fork: https://github.com/aliencaocao/OmniParser/commit/a3fe0e11c5b67a6ed8de916cdd77ddffba4d135f

**Comment by nmstoker** [comment]
Thanks for sharing that!

I'll try it out.

Incidentally I agree switching to a better OCR seems best but for anyone after a quick fix in the Gradio demo, I found that bumping the value for **text_threshold** in **easyocr_args** down from **0.9** to **0.7** or so helped reduce a fair amount of missing or prematurely split text.

https://github.com/microsoft/OmniParser/blob/d7708e830a2f16a40a8970211a9911fa776d4aa2/gradio_demo.py#L68

**Comment by aliencaocao** [comment]
Setting to 0.7 or 0.5 both did not make things better for this particular example:
![Image](https://github.com/user-attachments/assets/1dd507f0-ff92-4945-9245-7d4ea75fba2c)
I still think paddleocr is a clear better alternative.

**Comment by nmstoker** [comment]
Yes, I was agreeing with you 🙂

The quick fix isn't going to work for everything, but it did slightly improve several of my images where the full width of a text span wasn't being captured.

In your image above this happens with the timestamps: in your earlier version item 31 is missing the 2022 plus five entries under the Solved and Points columns were missed and above only one is missed. However it's still not handling the entries in the Rank column well enough.

**Comment by yadong-lu** [comment]
> Yes, I can make a PR if you wish, I already have it impl in my fork: [aliencaocao@a3fe0e1](https://github.com/aliencaocao/OmniParser/commit/a3fe0e11c5b67a6ed8de916cdd77ddffba4d135f)

Feel free to make a PR

**Comment by aliencaocao** [comment]
> > Yes, I can make a PR if you wish, I already have it impl in my fork: [aliencaocao@a3fe0e1](https://github.com/aliencaocao/OmniParser/commit/a3fe0e11c5b67a6ed8de916cdd77ddffba4d135f)
> 
> Feel free to make a PR

PR made
